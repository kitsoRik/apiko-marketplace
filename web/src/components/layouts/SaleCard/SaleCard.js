import React from "react";

import "./SaleCard.scss";

const SaleCard = ({
	id,
	product = { title: "" },
	user = { fullName: "" },
	date,
}) => {
	if (!user) throw new Error();
	const dateString = new Date(1970, 0, 1);

	dateString.setMilliseconds(date);

	return (
		<div className="sale-card">
			<div className="sale-card-icon">
				<img
					src="https://image.shutterstock.com/image-vector/pagination-bars-web-buttons-vector-260nw-1111694900.jpg"
					alt="Product"
				/>
			</div>
			<span>{product.title}</span>
			<span>{dateString.toString()}</span>
		</div>
	);
};

export default SaleCard;
