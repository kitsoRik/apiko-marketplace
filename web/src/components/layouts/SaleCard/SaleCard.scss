.sale-card {
    display: grid;
    grid-template-columns: min-content auto;
    grid-template-rows: min-content auto;

    border: 1px solid #349A89;

    &-icon {
        height: 64px;

        grid-row: 1 / span 2;

        & > img{
            max-height: 100%;
        }
    }
}