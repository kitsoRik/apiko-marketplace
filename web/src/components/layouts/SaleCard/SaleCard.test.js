import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import SaleCard from './SaleCard';


describe("Label", () => {
    it("should be rendered", () => {
        const component = create(<SaleCard />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
    it("should throw error with product=null", () => {
        const component = () => mount(<SaleCard product={null} />);

        expect(component).toThrowError();
    });
    it("should throw error with user=null", () => {
        const component = () => mount(<SaleCard user={null} />);

        expect(component).toThrowError();
    });
});