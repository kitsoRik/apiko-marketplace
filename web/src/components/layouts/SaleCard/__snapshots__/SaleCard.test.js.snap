// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should be rendered 1`] = `
<div
  className="sale-card"
>
  <div
    className="sale-card-icon"
  >
    <img
      alt="Product"
      src="https://image.shutterstock.com/image-vector/pagination-bars-web-buttons-vector-260nw-1111694900.jpg"
    />
  </div>
  <span>
    
  </span>
  <span>
    Invalid Date
  </span>
</div>
`;

exports[`Label should be rendered 2`] = `
<div
  className="sale-card"
>
  <div
    className="sale-card-icon"
  >
    <img
      alt="Product"
      src="https://image.shutterstock.com/image-vector/pagination-bars-web-buttons-vector-260nw-1111694900.jpg"
    />
  </div>
  <span>
    
  </span>
  <span>
    Invalid Date
  </span>
</div>
`;
